src/index.tsx

```diff
@@ -392,14 +392,14 @@ export class App extends React.Component<any, AppState> {
         const data = history.redoOnce();
         if (data !== null) {
           elements = data.elements;
-          this.setState(data.appState);
+          this.setState({ ...data.appState, elementType: "selection" });
         }
       } else {
         // undo action
         const data = history.undoOnce();
         if (data !== null) {
           elements = data.elements;
-          this.setState(data.appState);
+          this.setState({ ...data.appState, elementType: "selection" });
```

Comment:
Any specific reason why resetting to selection? This has the effect that when you have a shape locked, draw some shapes and undo one of them because you made a mistake, the tool is reset to selection, while arguably we should keep the shape tool you have locked.

Reply:
This was necessary because of multi arrow. Let's say I am in the middle of drawing arrows. If I undo, the arrow is removed but because multi-arrow tool doesn't work like the average shape tool (unlike dragging, multi tool stays in the tool until tool is changed or ESC/Enter is clicked), there needs to be a way to revert it back to select tool.

What if we keep the active tool in history? So, when you undo, you will get the shape that was being used (not sure if it is useful but still).

Reply:
Makes me think we should actually ignore undo/redo during `draggingElement`, `resizingElement`, or `multiElement` because we'll often end up in weird state otherwise (not just related to arrows).

Reply:
Better yet, also `editingElement`, thus:

```
else if (
      event[KEYS.META] &&
      event.code === "KeyZ" &&
      !this.state.resizingElement &&
      !this.state.editingElement &&
      !this.state.draggingElement &&
      !this.state.multiElement
    ) {
```

Reply:
Ultimately, undoing during `multiElement` should undo the last step in the arrow, but that's for another PR.

Reply:
I disabled it for `resizingElement`, `multiElement, and `editingElement`. `draggingElement`'s value is sometimes an Object (not sure why); so, I didn't add it to the comparisons. Additionally, I added a way to undo/redo parts of the arrow. So, if you have a multi arrow, it will remove each point instead of the whole arrow during undo.

Reply:
Yea, we should fix the `draggingElement`, but in another PR.